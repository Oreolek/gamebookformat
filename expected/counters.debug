BEGIN DEBUG OUTPUT
Book title: Counters Example
Number of sections: 9
Introduction
 This example gamebook shows how to use counters.


Turn to 1 to begin.
1 (start) -  This is where the gamebook starts. We have counters for [COUNT life]Life Points[/COUNT] and for [COUNT gold]Gold[/COUNT]. You start the adventure with [SET life]10[/SET] Life Points and [SET gold]12[/SET] Gold. You can never have less than [MIN gold]0[/MIN] Gold. FIXME need to figure out best syntax for death when life points go below 1. Go to 9 to get more Gold or 8 to lose some Life Points. Go to 3 for testing the init value markup. 
2 (inittest2) -  OK, we increase that something counter by [INC something]1[/INC] here, and also increase the other counter by [INC other]1[/INC]. You can go back to see the difference between set and init here (3), stuck in an infinite loop.
3 (inittest) -  The [COUNT something]Something Counter[/COUNT] starts at [INIT something]5[/SET], but when you come back here it will not reset to that, unlike the [COUNT other]Other Counter[/COUNT] that will reset to [SET other]5[/SET] every time you get here. If we try to init gold to [INIT gold]999[/SET] nothing will happen because it was already set to something before you came yere. Go on to 2. 
4 (death) -  Less than 1 Life Point means the adventure ends now. 
5 (hurrah) -  You have at least 16 gold. Go back to get some more at 9. 
6 (rich) -  You have more than 16 Gold. How nice. Go back to get some more at 9. 
7 (losegold) -  You drop [DEC gold]5[/DEC] Gold. It should not be possible to go below 0. Then go on to get gold at 9 or lose life at 8. 
8 (dangerous) -  You lose [DEC life]1[/DEC] Life Points. You can go to 9 to get some Gold. 
9 (getgold) -  Congratulations, you found [INC gold]2[/INC] Gold. If you have less than 1 Life Points, turn to ((automatic))4. You can go to 8 to lose some life or to 7 to drop some gold. If you have at least 16 Gold you can turn to 5. If you have even more than 16 Gold you can turn to 6. 
END DEBUG OUTPUT
