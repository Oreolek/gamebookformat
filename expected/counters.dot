digraph gamebook {

 Life PointsGold101201->7
1->6
3->7
4->7
55->7
5->6
16->7
217->2
7->6
7->5
167->3
167->4
}
