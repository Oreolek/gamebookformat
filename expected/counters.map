{"death": 4, "start": 1, "inittest2": 2, "inittest": 3, "getgold": 9, "losegold": 7, "rich": 6, "dangerous": 8, "hurrah": 5}