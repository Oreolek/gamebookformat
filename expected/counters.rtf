{\rtf1\ansi\ansicpg1252\cocoartf1038\cocoasubrtf360
{\fonttbl\f0\fswiss\fcharset0 Helvetica;}
{\colortbl;\red255\green255\blue255;}
\paperw11900\paperh16840\margl1440\margr1440\vieww14140\viewh14860\viewkind0
\pard\tx566\tx1133\tx1700\tx2267\tx2834\tx3401\tx3968\tx4535\tx5102\tx5669\tx6236\tx6803\ql\qnatural\pardirnatural

\f0\b\fs24 \cf0

\b \qc Counters Example
\b0\
\
\b \qc Introduction
\b0\



\ql  This example gamebook shows how to use counters. \
\

\b Turn to 1 to begin.
\b0\
\
\b \qc 1
\b0\
\ql  This is where the gamebook starts. We have counters for Life Points and for Gold. You start the adventure with 10 Life Points
 and 12 Gold
. You can never have less than 0 Gold
. FIXME need to figure out best syntax for death when life points go below 1. Go to \b 7
\b0
 to get more Gold or \b 6
\b0
 to lose some Life Points.  \
\

\b \qc 2
\b0\
\ql  Less than 1 Life Point means the adventure ends now. \
\

\b \qc 3
\b0\
\ql  You have at least 16 gold. Go back to get some more at \b 7
\b0
.  \
\

\b \qc 4
\b0\
\ql  You have more than 16 Gold. How nice. Go back to get some more at \b 7
\b0
.  \
\

\b \qc 5
\b0\
\ql  You drop 5 Gold
. It should not be possible to go below 0. Then go on to get gold at \b 7
\b0
 or lose life at \b 6
\b0
.  \
\

\b \qc 6
\b0\
\ql  You lose 1 Life Points
. You can go to \b 7
\b0
 to get some Gold.  \
\

\b \qc 7
\b0\
\ql  Congratulations, you found 2 Gold
. If you have less than 1, turn to \b 2
\b0
. You can go to \b 6
\b0
 to lose some life or to \b 5
\b0
 to drop some gold. If you have at least 16 you can turn to \b 3
\b0
. If you have even more than 16 you can turn to \b 4
\b0
.  \
\

}
