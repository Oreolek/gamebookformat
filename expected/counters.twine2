<tw-storydata name="Counters Example" startnode="1" creator="Twine" creator-version="2.0" format="Harlowe" options="">
<style role="stylesheet" id="twine-user-stylesheet" type="text/twine-css"></style><script role="script" id="twine-user-script" type="text/twine-javascript"></script>
<tw-passagedata pid="1" name="1" tags="" position="10,10"> This is where the gamebook starts. We have counters for *Life Points* and for *Gold*. You start the adventure with *10* Life Points and *12* Gold. You can never have less than *0* Gold. FIXME need to figure out best syntax for death when life points go below 1. Go to [[9]]
 to get more Gold or [[8]]
 to lose some Life Points. Go to [[3]]
 for testing the init value markup. </tw-passagedata>
<tw-passagedata pid="2" name="2" tags="" position="20,20"> OK, we increase that something counter by *1* here, and also increase the other counter by *1*. You can go back to see the difference between set and init [[here->3]]
, stuck in an infinite loop.</tw-passagedata>
<tw-passagedata pid="3" name="3" tags="" position="30,30"> The *Something Counter* starts at 5, but when you come back here it will not reset to that, unlike the *Other Counter* that will reset to *5* every time you get here. If we try to init gold to 999 nothing will happen because it was already set to something before you came yere. Go on to [[2]]
. </tw-passagedata>
<tw-passagedata pid="4" name="4" tags="" position="40,40"> Less than 1 Life Point means the adventure ends now. </tw-passagedata>
<tw-passagedata pid="5" name="5" tags="" position="50,50"> You have at least 16 gold. Go back to get some more at [[9]]
. </tw-passagedata>
<tw-passagedata pid="6" name="6" tags="" position="60,60"> You have more than 16 Gold. How nice. Go back to get some more at [[9]]
. </tw-passagedata>
<tw-passagedata pid="7" name="7" tags="" position="70,70"> You drop *5* Gold. It should not be possible to go below 0. Then go on to get gold at [[9]]
 or lose life at [[8]]
. </tw-passagedata>
<tw-passagedata pid="8" name="8" tags="" position="80,80"> You lose *1* Life Points. You can go to [[9]]
 to get some Gold. </tw-passagedata>
<tw-passagedata pid="9" name="9" tags="" position="90,90"> Congratulations, you found *2* Gold. If you have less than *1* Life Points, turn to [[4]]
. You can go to [[8]]
 to lose some life or to [[7]]
 to drop some gold. If you have at least *16* Gold you can turn to [[5]]
. If you have even more than *16* Gold you can turn to [[6]]
. </tw-passagedata>
</tw-storydata>
