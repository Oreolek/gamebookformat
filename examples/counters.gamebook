title = Counters Example
= Introduction
This example gamebook shows how to use counters.
* 1 start
This is where the gamebook starts.
We have counters for [count life]Life Points[/count] and for
[count gold]Gold[/count]. You start the adventure with
[set life]10[/set] and [set gold]12[/set].
You can never have less than [min gold]0[/min]. FIXME need to
figure out best syntax for death when life points go below 1.
Go to [[getgold]] to get more Gold or [[dangerous]] to lose some Life Points.

* getgold
Congratulations, you found [inc gold]2[/gold].
You can go to [[dangerous]] to lose some life or to [[losegold]] to drop
some gold.

* dangerous
You lose [dec life]1[/dec].
You can go to [[getgold]] to get some Gold.

* losegold
You drop [dec gold]5[/dec]. It should not be possible to
go below 0. Then go on to get gold at [[getgold]] or lose life at [[dangerous]].
