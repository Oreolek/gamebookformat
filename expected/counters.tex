\documentclass[a5paper,onecolumn]{book}

\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage[hidelinks]{hyperref}
\usepackage{graphicx}

\usepackage[top=3.3cm, bottom=3.3cm, left=2cm, right=2cm]{geometry}
\newif\ifpdf
\ifx\pdfoutput\undefined
  \pdffalse
\else
  \ifnum\pdfoutput=1
    \pdftrue
  \else
    \pdffalse
  \fi
\fi

\title{Counters Example}
\author{}
\date{}

\newcounter{sectionnr}

\begin{document}

\maketitle

\thispagestyle{empty}

\pagestyle{empty}

\clearpage

\subsection*{\begin{center} \textbf{Introduction} \end{center}}


 \noindent
 This example gamebook shows how to use counters.
\vspace{1em}


Turn to 1 to begin.
\phantomsection
\refstepcounter{sectionnr}
\label{section1}
\subsection*{\begin{center} \textbf{1} \end{center}}

 \noindent
 This is where the gamebook starts. We have counters for Life Points and for Gold. You start the adventure with 10 Life Points and 12 Gold. You can never have less than 0 Gold. FIXME need to figure out best syntax for death when life points go below 1. Go to \textbf{\autoref{section9}} to get more Gold or \textbf{\autoref{section8}} to lose some Life Points. Go to \textbf{\autoref{section3}} for testing the init value markup. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section2}
\subsection*{\begin{center} \textbf{2} \end{center}}

 \noindent
 OK, we increase that something counter by 1 here, and also increase the other counter by 1. You can go back to see the difference between set and init here (\textbf{\autoref{section3}})
, stuck in an infinite loop.
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section3}
\subsection*{\begin{center} \textbf{3} \end{center}}

 \noindent
 The Something Counter starts at 5, but when you come back here it will not reset to that, unlike the Other Counter that will reset to 5 every time you get here. If we try to init gold to 999 nothing will happen because it was already set to something before you came yere. Go on to \textbf{\autoref{section2}}. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section4}
\subsection*{\begin{center} \textbf{4} \end{center}}

 \noindent
 Less than 1 Life Point means the adventure ends now. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section5}
\subsection*{\begin{center} \textbf{5} \end{center}}

 \noindent
 You have at least 16 gold. Go back to get some more at \textbf{\autoref{section9}}. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section6}
\subsection*{\begin{center} \textbf{6} \end{center}}

 \noindent
 You have more than 16 Gold. How nice. Go back to get some more at \textbf{\autoref{section9}}. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section7}
\subsection*{\begin{center} \textbf{7} \end{center}}

 \noindent
 You drop 5 Gold. It should not be possible to go below 0. Then go on to get gold at \textbf{\autoref{section9}} or lose life at \textbf{\autoref{section8}}. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section8}
\subsection*{\begin{center} \textbf{8} \end{center}}

 \noindent
 You lose 1 Life Points. You can go to \textbf{\autoref{section9}} to get some Gold. 
\vspace{1em}
\phantomsection
\refstepcounter{sectionnr}
\label{section9}
\subsection*{\begin{center} \textbf{9} \end{center}}

 \noindent
 Congratulations, you found 2 Gold. If you have less than 1 Life Points, turn to \textbf{\autoref{section4}}. You can go to \textbf{\autoref{section8}} to lose some life or to \textbf{\autoref{section7}} to drop some gold. If you have at least 16 Gold you can turn to \textbf{\autoref{section5}}. If you have even more than 16 Gold you can turn to \textbf{\autoref{section6}}. 
\vspace{1em}
\end{document}
